src/Devices.cpp

```diff
@@ -0,0 +1,101 @@
+#include "yacx/Devices.hpp"
+
+#include "yacx/Exception.hpp"
+#include "yacx/Init.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <sstream>
+
+using yacx::Device, yacx::Devices;
+
+Devices *Devices::instance = NULL;
+
+Devices *Devices::getInstance() {
+  if (instance == NULL) {
+    instance = new Devices();
```

Comment:
shouldn't this also be deleted in the destructor?

Reply:
Ich weiß jetzt auch nicht wie man deleten sollte. `instance` ist halt so ein statisches Attribut. Deshalb weiß ich nicht, ob es so sinnvoll ist dieses statische Attribut im Desktrutor von Devices zu deleten.
So wie es jetzt ist wird es einfach gar nicht gefreet

Reply:
Hmm okay, denke das ist nicht so schlimm, aber vielleicht ein Issue fuer spaeter erstellen lol